Compilation error with latest version (but not with previous one)

Getting a compilation error on this code,  This is as small as I was able to make it while getting the same compilation error. 
```
#pythran export test()
import numpy as np


def A_I():
    s = np.array([0.70817816, 0.68863678], dtype=np.float64)
    m = np.array([-1.11312199, -0.99629629], dtype=np.float64)
    self = tuple([100, s, m])
    return self


def A_F1(self, Input_x):
    Input_x = (Input_x - self[2]) / self[1]
    return Input_x

def B_I():
    """Constructor for B"""
    self = tuple([0.0, 0])
    return self


def C_I(sRate):
    """Constructor for class """
    a = A_I()
    b = B_I()
    self = tuple([0, sRate, 0, a, b])
    return self


def C_Test(self):
    F = np.zeros((1, 1, 1))
    ret = A_F1(self[3], F.astype(np.float32))
    print(ret[0, 0])

def test():
    TD = C_I(44100)
    C_Test(TD)

```
The error is 
```
In file included from /Users/jlaroche/ua_conda_env-Darwin.x86_64/lib/python3.8/site-packages/pythran/pythonic/core.hpp:29:
In file included from /Users/jlaroche/ua_conda_env-Darwin.x86_64/lib/python3.8/site-packages/pythran/pythonic/types/slice.hpp:5:
In file included from /Users/jlaroche/ua_conda_env-Darwin.x86_64/lib/python3.8/site-packages/pythran/pythonic/types/NoneType.hpp:8:
In file included from /Users/jlaroche/ua_conda_env-Darwin.x86_64/lib/python3.8/site-packages/pythran/pythonic/builtins/bool_.hpp:4:
In file included from /Users/jlaroche/ua_conda_env-Darwin.x86_64/lib/python3.8/site-packages/pythran/pythonic/include/builtins/bool_.hpp:5:
/Users/jlaroche/ua_conda_env-Darwin.x86_64/lib/python3.8/site-packages/pythran/pythonic/include/types/tuple.hpp:795:60: error: pack expansion contains parameter packs 't0' and 't1' that have
      different lengths (3 vs. 2)
  using type = std::tuple<typename __combined<t0, t1>::type...>;
                                              ~~  ~~       ^
```

--- a/pythran/pythonic/include/types/tuple.hpp
+++ b/pythran/pythonic/include/types/tuple.hpp
@@ -805,15 +805,29 @@ struct __combined<container<t>, std::tuple<t0...>> {
   using type = std::tuple<t0...>;
 };
 
+PYTHONIC_NS_BEGIN
+namespace details
+{
+  template <class T, class P, bool Same>
+  struct pick_combined;
+  template <class T, class P>
+  struct pick_combined<T, P, true> {
+    using type = typename __combined<T, P>::type;
+  };
+  template <class T, class P>
+  struct pick_combined<T, P, false> {
+    using type = P;
+  };
+}
+PYTHONIC_NS_END
+
 template <long I, class t, class... t0>
 struct __combined<std::tuple<t0...>,
                   indexable_container<std::integral_constant<long, I>, t>> {
   using holder = std::tuple<t0...>;
   template <size_t... Is>
-  static std::tuple<typename std::conditional<
-      I == Is, typename __combined<
-                   t, typename std::tuple_element<Is, holder>::type>::type,
-      typename std::tuple_element<Is, holder>::type>::type...>
+  static std::tuple<typename pythonic::details::pick_combined<
+      t, typename std::tuple_element<Is, holder>::type, I == Is>::type...>
       make_type(pythonic::utils::index_sequence<Is...>);
   static auto make_type() -> decltype(
       make_type(pythonic::utils::make_index_sequence<sizeof...(t0)>()));

--- a/pythran/tests/test_advanced.py
+++ b/pythran/tests/test_advanced.py
@@ -446,3 +446,38 @@ def global_effects_partial0(l):
         def annotations(x: np.ndarray) -> np.ndarray:
             return x'''
         self.run_test(code, numpy.ones(1), annotations=[NDArray[float, :]])
+
+    def test_tuple_indexable_container(self):
+        code = """
+import numpy as np
+def A_I():
+    s = np.array([0.70817816, 0.68863678], dtype=np.float64)
+    m = np.array([-1.11312199, -0.99629629], dtype=np.float64)
+    self = tuple([100, s, m])
+    return self
+
+def A_F1(self, Input_x):
+    Input_x = (Input_x - self[2]) / self[1]
+    return Input_x
+
+def B_I():
+    self = tuple([0.0, 0])
+    return self
+
+
+def C_I(sRate):
+    a = A_I()
+    b = B_I()
+    self = tuple([0, sRate, 0, a, b])
+    return self
+
+
+def C_Test(self):
+    F = np.zeros((1, 1, 1))
+    ret = A_F1(self[3], F.astype(np.float32))
+    return ret
+
+def tuple_indexable_container(n):
+    TD = C_I(n)
+    return C_Test(TD)"""
+        self.run_test(code, 44100, tuple_indexable_container=[int])

